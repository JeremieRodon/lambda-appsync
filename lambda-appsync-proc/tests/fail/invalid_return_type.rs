use lambda_appsync::{appsync_lambda_main, appsync_operation, AppsyncError};

appsync_lambda_main!("../../../../schema.graphql");

// Wrong return type (String instead of Player)
#[appsync_operation(query(players))]
async fn wrong_return_type() -> Result<String, AppsyncError> {
    Ok("not a Player vector".to_string())
}
