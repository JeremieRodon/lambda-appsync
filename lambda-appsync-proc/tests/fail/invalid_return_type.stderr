error[E0271]: expected `wrong_return_type` to be a fn item that returns `Result<Vec<Player>, AppsyncError>`, but it returns `Result<String, AppsyncError>`
 --> tests/fail/invalid_return_type.rs:7:10
  |
6 | #[appsync_operation(query(players))]
  | ------------------------------------ required by a bound introduced by this call
7 | async fn wrong_return_type() -> Result<String, AppsyncError> {
  |          ^^^^^^^^^^^^^^^^^ expected `Result<Vec<Player>, AppsyncError>`, found `Result<String, AppsyncError>`
  |
  = note: expected enum `Result<Vec<Player>, AppsyncError>`
             found enum `Result<std::string::String, AppsyncError>`
note: required by a bound in `players::without_event::check_signature`
 --> tests/fail/invalid_return_type.rs:3:22
  |
3 | appsync_lambda_main!("../../../../schema.graphql");
  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `check_signature`
